library/footer.cpp

```diff
@@ -0,0 +1,289 @@
+#include "footer.h"
+
+#define __STDC_FORMAT_MACROS
+#include <inttypes.h>
+
+#include "blob.h"
+#include "crypto/sha512.h"
+#include "crc32.h"
+
+#include "measure_points.h"
+
+static const uint64_t EBLOB_CSUM_CHUNK_SIZE = 1UL<<20;
+
+/*
+ * eblob_disk_footer contains csum of data.
+ * @csum - sha512 of record's data.
+ *
+ * eblob_disk_footer are kept at the end of the recods.
+ */
+struct eblob_disk_footer {
+	unsigned char	csum[EBLOB_ID_SIZE];
+	uint64_t	offset;
+} __attribute__ ((packed));
+
+/*
+ * crc32_file() - computes crc32 of bytes range read from @fd with @offset and @count.
+ *
+ * Results:
+ * Returns negative error value or zero on success.
+ * @result - computed crc32.
+ */
+static inline int crc32_file(int fd, off_t offset, size_t count, uint32_t &result) {
+	static const size_t buffer_size = 1024;
+	char buffer[buffer_size];
+	size_t read_size = buffer_size;
+	int err = 0;
+	result = 0;
+
+	while (count) {
+		if (count < buffer_size)
+			read_size = count;
+
+		err = __eblob_read_ll(fd, buffer, read_size, offset);
+		if (err)
+			break;
+
+		result = crc32_buffer(buffer, read_size, result);
+		count -= read_size;
+		offset += read_size;
+	}
+
+	return err;
+}
+
+/*
+ * eblob_get_footer_offset() - calculates footer offset within record pointed by @wc.
+ *
+ * Returns footer offset within record.
+ */
+static inline uint64_t eblob_get_footer_offset(struct eblob_write_control *wc) {
+	if (wc->flags & BLOB_DISK_CTL_CHUNKED_CRC32) {
+		static const size_t f_size = sizeof(uint32_t);
+		/* size of whole record without header and final checksum */
+		const uint64_t size = wc->total_size - sizeof(struct eblob_disk_control) - f_size;
+		/* number of chunks */
+		const uint64_t chunks_count = ((size  - 1) / (EBLOB_CSUM_CHUNK_SIZE + f_size)) + 1;
```

Comment:
> When we determine number of chunks, we have to divide second line into (EBLOB_CSUM_CHUNK_SIZE + f_size) parts.

Also we have to round up result because last chunk_X can be less than EBLOB_CSUM_CHUNK_SIZE.